Commands/Provider/SPOProvider.cs

```diff
@@ -631,14 +676,30 @@ private IEnumerable<object> GetFolderItems(Folder folder, bool throwError = fals
                         }
                     }
 
-                    //Get files and folders
-                    var files = folder.Context.LoadQuery(folder.Files).OrderBy(f => f.Name);
-                    var folders = folder.Context.LoadQuery(folder.Folders).OrderBy(f => f.Name);
-                    folder.Context.ExecuteQueryRetry();
+                    //If large document library use caml query
+                    folder.EnsureProperty(p => p.ItemCount);
+                    if (folder.ItemCount > 5000)
+                    {
+                        if (!limit.HasValue)
+                        {
+                            WriteWarning("Large document library! Only displaying the 100 first items. Use Get-ChildItems -Limit <int>|All");
+                        }
+                        WriteWarning("Large document library! Template folder and files will not be shown.");
+
+                        folderAndFiles.AddRange(limit.HasValue ? GetListItems(folder, limit.Value) : GetListItems(folder));
+                    }
+                    else
+                    {
+                        //Get files and folders
+                        var files = folder.Context.LoadQuery(folder.Files.Include(f => f.Name, f => f.ServerRelativeUrl, f => f.TimeLastModified, f => f.Length)).OrderBy(f => f.Name);
+                        var folders = folder.Context.LoadQuery(folder.Folders.Include(f => f.Name, f => f.ItemCount, f => f.TimeLastModified, f=>f.ServerRelativeUrl)).OrderBy(f => f.Name);
```

Comment:
When I build this in Debug15 mode i get the error:

> Error	CS1061	'Folder' does not contain a definition for 'TimeLastModified' and no extension method 'TimeLastModified' accepting a first argument of type 'Folder' could be found (are you missing a using directive or an assembly reference?)	SharePointPnP.PowerShell.Commands	C:\Users\SP-Install\Source\Repos\PnP-PowerShell\Commands\Provider\SPOProvider.cs	700	Active

@VesaJuvonen @afsandeberg Has this been tested for SharePoint 2013 on-prem?  When i build the Debug or Debug16 configuration, it does work.

Also when I have a look in the different version of .dll's i can see the Debug15 doesn't support the 'TimeLastModified' property and the .dll that is used by the Debug configuration does have the property.
![image](https://cloud.githubusercontent.com/assets/22449250/23207928/6c0ac7e2-f8f3-11e6-99c7-24bd6009385a.png)
vs
![image](https://cloud.githubusercontent.com/assets/22449250/23207866/19de32b0-f8f3-11e6-9b2c-5c53410b6e5a.png)